test/test_mps.py

```diff
@@ -6630,6 +6577,39 @@ def helper(input_shape, out_shape, channels_last):
         except Exception as e:
             pass
 
+    @parametrize(
+        "input_shape,output_shape",
+        [
+            ((1, 1, 4, 4), (2, 2)),
+            ((1, 1, 3, 3), (2, 2)),
+            ((1, 2, 2, 3), (5, 4)),
+            ((1, 64, 10, 9), (None, 7)),
+        ],
+    )
+    @parametrize("channels_last", [False, True])
+    def test_adaptive_avg_pool2d_nondivisible_sizes(self, input_shape, output_shape, channels_last):
```

Comment:
I believe this is already tested in the OpInfo db, so you should be able to remove the entries for `adaptive_avg_pool2d` and `adaptive_avg_pool1d` from `UNDEFINED_XFAILLIST` in `torch/testing/_internal/common_mps.py` and then you could remove this test